### PR TITLE
Script Project Generation

### PR DESCRIPTION
**Pull request by TejasJD**
- Creating a new Project now generates a script project (Visual studio C#) solution along with it.
- Added FileWatch header file, which would be used for next steps of the system.

### CODE REVIEW

File: HorizonEngine/src/HorizonEngine/Scripting/ScriptEngine.cpp [hunk 1]
```diff
@@ -104,45 +125,53 @@ namespace Hzn
...
 
+	void ScriptEngine::LoadAppAssembly(const std::filesystem::path& path)
+	{
+		s_Data->appAssembly = Utils::LoadCSharpAssembly(path.string());
+		s_Data->appAssemblyImage = mono_assembly_get_image(s_Data->appAssembly);
+	}
+
+
 
 	void ScriptEngine::init()
```
Comment (LiamANeeson) [inline HorizonEngine/src/HorizonEngine/Scripting/ScriptEngine.cpp:136]: Will commented code be added in again when fully implemented?

**Review by alex-mulkerrins — APPROVED** [review]

**Review by BozhidarVelikov — APPROVED** [review]

File: HorizonEditor/CMakeLists.txt [hunk 2]
```diff
@@ -78,4 +78,22 @@ add_custom_command(
...
+add_custom_command(
+	TARGET HznEditor
+	POST_BUILD
+	COMMAND ${CMAKE_COMMAND}
+	ARGS -E copy
+	${CMAKE_CURRENT_LIST_DIR}/premake5.exe
+	$<TARGET_FILE_DIR:HznEditor>/premake5.exe
+)
+
+add_custom_command(
```
Comment (LiamANeeson) [inline HorizonEditor/CMakeLists.txt:92]: Adding this block in twice? Why?

File: HorizonEditor/src/EditorLayer.cpp [hunk 3]
```diff
@@ -48,47 +48,49 @@ void EditorLayer::onDetach()
 
 void EditorLayer::onUpdate(Hzn::TimeStep ts)
 {
-	m_FrameBuffer->bind();
+	if (Hzn::SceneManager::isOpen())
+	{
+		m_FrameBuffer->bind();
+		// here, in case if the framebuffer is re-created, and the last known
+		// viewport size does not match the viewport size of the new framebuffer, then
+		// we update all the camera components to the proper aspect ratio, and update the last known viewport size.
```
Comment (LiamANeeson) [inline HorizonEditor/src/EditorLayer.cpp:56]: Makes sense, thanks for comment

**Review by Minghui-Gao — APPROVED** [review]

File: HorizonEditor/CMakeLists.txt (hunk 2 above)
Comment (TejasJD) [reply HorizonEditor/CMakeLists.txt:92]: One is for the script file, and the other is for the premake application that runs the script file.

File: HorizonEditor/src/EditorLayer.cpp (hunk 3 above)
Comment (TejasJD) [reply HorizonEditor/src/EditorLayer.cpp:56]: Okay.

**Pull request merged**